File: jiaoji_app_back/src/main/java/com/example/jiaoji_app_back/entity/ActivityDetails.java
/*author: qyl*/
package com.example.jiaoji_app_back.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor

@Table(name ="activity_details")
public class ActivityDetails {


    @Id
    private Long id;
    @Basic
    @Column(name = "name")
    private String name;
    @Basic
    @Column(name = "content")
    private String content;
    @Basic
    @Column(name = "location")
    private String location;
    @Basic
    @Column(name = "signupTime")

    private String signupTime;
    @Basic
    @Column(name = "activityTime")
    private String activityTime;
    @Basic
    @Column(name = "departments")
    private String departments;
    @Basic
    @Column(name = "signupRestriction")
    private String signupRestriction;
    @Basic
    @Column(name = "college")
    private String college;
    @Basic
    @Column(name = "grade")
    private String grade;
    @Basic
    @Column(name = "club")
    private String club;
    @Basic
    @Column(name = "recruitmentNumber")
    private Long recruitmentNumber;
    @Basic
    @Column(name = "remainingNumber")
    private Long remainingNumber;
    @Basic
    @Column(name = "organizer")
    private String organizer;
    @Basic
    @Column(name = "suScore")
    private Long suScore;
    @Basic
    @Column(name = "laborHour")
    private Long laborHour;
    @Basic
//
//    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private String status;
//    public void setStatus(Status status){
//        this.status = status;
//    }
    @Basic
    @Column(name = "comments")

    private String comments;
//    private String photo;

    public ActivityDetails(Long id, String name, String content, String location, String signupTime, String activityTime, String departments, String signupRestriction, String college, String grade, String club, Long recruitmentNumber, Long remainingNumber,String organizer, Long suScore, Long laborHour, String status, String comments) {
        this.id = id;
        this.name = name;
        this.content = content;
        this.location = location;
        this.signupTime = signupTime;
        this.activityTime = activityTime;
        this.departments = departments;
        this.signupRestriction = signupRestriction;
        this.college = college;
        this.grade = grade;
        this.club = club;
        this.recruitmentNumber = recruitmentNumber;
        this.remainingNumber = remainingNumber;
        this.organizer = organizer;
        this.suScore = suScore;
        this.laborHour = laborHour;
        this.status = status;
        this.comments = comments;
//        this.photo = photo;
    }


}

File: src/Component/ActivityDetails.js
import React from "react";
import { Card as AntCard } from "antd";
import "../css/ActivityDetail.css";
// import { Map, Marker, NavigationControl, InfoWindow } from 'react-bmap';

const ActivityDetails = ({ activity }) => {
    return (
        <div className="card-container">
            {/*<Map center={{lng: 121.449, lat: 31.029}} zoom="30">*/}
            {/*    <Marker position={{lng: 121.449, lat:31.029}} />*/}
            {/*    <InfoWindow position={{lng: 121.449, lat:31.029}} text="软件学院" title="活动地点"/>*/}
            {/*    <NavigationControl />*/}
            {/*</Map>*/}
            <AntCard className="activity-card ant-card-hoverable">
                <div className="ant-card-head">
                    <h2 className="ant-card-head-title">{activity && activity.name}</h2>
                </div>
                <div className="ant-card-body">
                    <p>{activity && activity.content}</p>
                    {
                        activity
                        &&
                        <ul>
                            <li>地点：{activity.location}</li>
                            <li>报名时间：{activity.signupTime}</li>
                            <li>活动时间：{activity.activityTime}</li>
                            <li>开设学院：{activity.departments}</li>
                            <li>报名限制：</li>
                            <li>学院限制：{activity.college || '无'}</li>
                            <li>年级限制：{activity.grade || '无'}</li>
                            <li>社团限制：{activity.club || '无'}</li>
                            <li>招募人数：{activity.recruitmentNumber}</li>
                            <li>剩余名额：{activity.remainingNumber}</li>
                            <li>主办单位：{activity.organizer}</li>
                            <li>素拓奖励：{activity.suScore} 分</li>
                            <li>劳动教育学时：{activity.laborHour} 时</li>
                        </ul>
                    }

                </div>
            </AntCard>
        </div>
    );
};

export default ActivityDetails;
